markets/perps-market/contracts/storage/PerpsAccount.sol

```diff
@@ -176,8 +176,14 @@ library PerpsAccount {
     function validateWithdrawableAmount(
         Data storage self,
         uint128 synthMarketId,
-        uint256 amountToWithdraw
+        uint256 amountToWithdraw,
+        ISpotMarketSystem spotMarket
     ) internal view returns (uint256 availableWithdrawableCollateralUsd) {
+        uint collateralAmount = self.collateralAmounts[synthMarketId];
+        if (collateralAmount < amountToWithdraw) {
+            revert InsufficientSynthCollateral(synthMarketId, collateralAmount, amountToWithdraw);
+        }
```

Comment:
moved this up since if they don't have enough of hte collateral, we should abort anyway before other checks